### ISSUE TITLE
[Bug]: LT02 flags the contents of ON CONFLICT ... DO UPDATE SET as over-indented (fixed upstream in sqlfluff)

### ISSUE THREAD

**Issue by okbear** [quarylabs/sqruff#4039]
sqruff reports `LT02 Line should not be indented` for the assignment list inside
`ON CONFLICT ... DO UPDATE SET`. sqlfluff does not report it, and in sqruff the
violation is also **unfixable** — `sqruff fix` leaves the file byte-identical, so
the only ways to reach a clean state are un-indenting the SET body (which hurts
readability) or a `noqa`.

### Case 1 — `ON CONFLICT` already at column 0

```sql
INSERT INTO t (a, b, c) VALUES (1, 2, 3)
ON CONFLICT (a) DO UPDATE SET
    b = excluded.b,
    c = excluded.c;
```

`sqruff lint --dialect postgres repro.sql` (0.40.0):

```
L:   3 | P:   1 | LT02 | Line should not be indented. [layout.indent]
L:   4 | P:   1 | LT02 | Line should not be indented. [layout.indent]
```

`sqlfluff lint --dialect postgres repro.sql` (4.3.0): `All Finished!` — no violations.

`sqruff fix --dialect postgres repro.sql` leaves the file unchanged, so the
reported violation cannot be fixed automatically.

### Case 2 — `ON CONFLICT` indented, which isolates the difference

```sql
INSERT INTO t (a, b, c)
SELECT 1 AS a, 2 AS b, 3 AS c
    ON CONFLICT (a) DO UPDATE SET
    b = excluded.b,
    c = excluded.c;
```

| line | sqruff 0.40.0 | sqlfluff 4.3.0 |
| --- | --- | --- |
| L3 `    ON CONFLICT (a) DO UPDATE SET` | LT02 | LT02 |
| L4 `    b = excluded.b,` | **LT02** | — |
| L5 `    c = excluded.c;` | **LT02** | — |

Both agree `ON CONFLICT` itself should not be indented (L3), but sqruff
additionally flags the SET assignment list. `sqlfluff fix` produces what I would
expect — `ON CONFLICT` dedented, SET body kept indented:

```sql
INSERT INTO t (a, b, c)
SELECT
    1 AS a,
    2 AS b,
    3 AS c
ON CONFLICT (a) DO UPDATE SET
    b = excluded.b,
    c = excluded.c;
```

### Expected behaviour

The SET assignment list should be allowed to be indented, matching sqlfluff.

### Upstream

This was reported upstream as sqlfluff#6659 ("Postgres `ON` preventing formatting
after occurrence") and is resolved there — 4.3.0 is clean on Case 1. The
discussion points at adding `Indent`/`Dedent` around the conflict action
(`ConflictActionSegment`), so this looks like an upstream fix that hasn't reached
sqruff's cursor yet rather than a sqruff-specific regression.

Possibly related open work: #3900 (indent update and returning clauses),
#3889 (LT02 oscillation), #3902 (templated LT02 edge cases), and #3320
(inconsistent lint vs fix for LT02 in another dialect).

### Version

```
sqruff 0.40.0 (darwin-aarch64, official release binary)
compared against sqlfluff 4.3.0
```
